src/controllers/admin/webhook.ts

```diff
@@ -43,69 +57,356 @@ export class WebhookController {
 			} satisfies INotifyMessage);
 			return response.sendStatus(StatusCodes.BAD_REQUEST);
 		}
-		// Handle the event
-		switch (event.type) {
-			case 'customer.subscription.trial_will_end':
-				subscription = event.data.object;
-				status = subscription.status;
-				await eventTracker.notify({
-					message: EventTracker.compileBasicNotification(
-						`Subscription status is ${status} for subscription with id: ${subscription.id}`,
-						'Stripe Webhook: customer.subscription.trial_will_end'
-					),
-					severity: 'info',
-				} satisfies INotifyMessage);
-				await eventTracker.submit(
-					buildSubmitOperation(subscription, OperationNameEnum.SUBSCRIPTION_TRIAL_WILL_END)
+
+		try {
+			// Handle the event
+			switch (event.type) {
+				case 'customer.subscription.trial_will_end':
+					subscription = event.data.object;
+					status = subscription.status;
+					await eventTracker.notify({
+						message: EventTracker.compileBasicNotification(
+							`Subscription status is ${status} for subscription with id: ${subscription.id}`,
+							'Stripe Webhook: customer.subscription.trial_will_end'
+						),
+						severity: 'info',
+					} satisfies INotifyMessage);
+					break;
+				case 'customer.subscription.deleted':
+					subscription = event.data.object;
+					status = subscription.status;
+					await eventTracker.notify({
+						message: EventTracker.compileBasicNotification(
+							`Subscription status is ${status} for subscription with id: ${subscription.id}`,
+							'Stripe Webhook: customer.subscription.deleted'
+						),
+						severity: 'info',
+					} satisfies INotifyMessage);
+					await this.handleSubscriptionCancel(subscription);
```

Comment:
A general style recommendation -
Since you're already handling the entire logic in these `handleSubscription<XXXX>` helper functions, it might be easier to just move the entire section inside the helper function and return from here?
- No need to declare a variable outside of switch block
- Notification can be already printed from inside the return
- You can just `return` instead of break and return

Reply:
@jay-dee7 added your first 2 suggestions. Leaving the third one as it is. As there are other parts of the code without response using the handlers